script.js

```diff
@@ -1,12 +1,137 @@
 //You can edit ALL of the code here
+const rootElem = document.getElementById("root");
+const dropDown = document.getElementById("list");
+let allEpisodes = getAllEpisodes();
+const searchBar = document.getElementById("search");
+const display = document.getElementById("display");
+const btn = document.getElementById("btn")
+let allShows = getAllShows();
+const showDropDown = document.getElementById("showList")
+let initialId = "82";
+let shows = [];
+
+let showId = allShows.map(el=> el.id);
+
+ function showDisplayer(showId){
+  fetch(`https://api.tvmaze.com/shows/${showId}/episodes`)
+  .then(response => response.json())
+  .then((data) =>{ shows = [...data] 
+  makePageForEpisodes(data)
+  allEpisodes = data
+  displayer()
+  dropDownSelection(data)})
+  .catch((error) => console.log(error));
+} 
+
+function makePageForEpisodes(episodeList) {
+  rootElem.innerHTML = "";
+  display.innerText = `Display: ${episodeList.length}/${allShows.length}`;
+  episodeList.forEach(e => {
+  const boxBody = document.createElement('div');
+  boxBody.id = "boxBody";
+  const title = document.createElement('h3');
+  title.id = "title";
+  const imgBox = document.createElement('img');
+  imgBox.id = "imgBox";
+  const para = document.createElement('p');
+  para.id = "para"
+  rootElem.appendChild(boxBody);
+  boxBody.append(title,imgBox,para);
+  title.innerText = `${e.name}`
+  +`-S${e.season.toString().padStart(2,'0')}E${e.number.toString().padStart(2,'0')}`;
+  imgBox.src = e.image.medium;
+  para.innerHTML = e.summary;
+});
+}
+
+//setting the filterd search results 
+function episodesFilter() {
+  searchBar.addEventListener("keyup", function(p){
+    let filteredValue = p.target.value;
+   let filteredList = allEpisodes.filter((result)=>{
+     let namee = result.name;
+     let text = result.summary;
+    return namee.toLowerCase().includes(filteredValue.toLowerCase())||text.toLowerCase().includes(filteredValue.toLowerCase())
+    })
+    makePageForEpisodes(filteredList);
+  });
+}
+
+ //display the Episodes on the selector menu/dropDown
+ function displayer() {
+   dropDown.innerHTML = "";
+  allEpisodes.forEach(element => {
+    let options = document.createElement('option');
+    options.innerText = `S${element.season.toString().padStart(2,'0')}E${element.number.toString().padStart(2,'0')}-${element.name}`;
+    dropDown.appendChild(options)
+  });
+ }
+
+
+// episode selector/picker
+ dropDown.addEventListener("change",dropDownSelection)
+function dropDownSelection(selector){
+selector = event.target.value
```

Comment:
You are doing something a bit funny here :)
If you are using `selector` as the function parameter, there is no point immediately reassigning something else to it - you will just lose the value that is passed in.
The reason that you don't get an error is that `event` is globally defined - but this is not actually the event you want. You want the `change` event that is passed to your function as a parameter.

This would work:
```
function dropDownSelection(event){
selector = event.target.value
```

Reply:
One of the errors in the console comes from here. The problem is that you are calling `dropDownSelection` on the initial page load, but at that point there is no `change event` to refer to, and hence `event.target.value` will give an error.

You can add a condition to catch that case, and just display the first episode:
```
if(!event || !event.target) {
  return makePageForEpisodes(allEpisodes.filter((episode, index) => index === 0))
}
```

So overall the start of this function would look like:
```
function dropDownSelection(event){

if(!event || !event.target) {
  return makePageForEpisodes(allEpisodes.filter((episode, index) => index === 0))
}

selector = event.target.value
```